src/dotnet/ShellShim/ZshDetector.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.ShellShim
+{
+    internal class ZshDetector
+    {
+        private readonly IEnvironmentProvider _environmentProvider;
+        private const string ZshFileName = "zsh";
+
+        public ZshDetector(IEnvironmentProvider environmentProvider = null)
+        {
+            _environmentProvider = environmentProvider;
+        }
+
+        /// <summary>
+        /// Returns true if the `SHELL` environment variable ends with `zsh` for the filename.
+        ///     By detecting $SHELL's value
```

Comment:
This line can now be deleted.  I don't like that suggestions can't be multiline (afaik).

Reply:
deleted the extra line